Lib/fractions.py

```diff
@@ -57,22 +57,42 @@ def _gcd(a, b):
 """, re.VERBOSE | re.IGNORECASE)
 
 
+def _as_integer_ratio(obj):
+    """
+    Return a pair (numerator, denominator) if obj has an ``as_integer_ratio``
+    method or is an instance of ``numbers.Rational``. Return ``NotImplemented``
+    if neither works.
+    """
+    try:
+        f = obj.as_integer_ratio
+    except AttributeError:
+        pass
+    else:
+        return f()
+
+    if isinstance(obj, numbers.Rational):
+        return (obj.numerator, obj.denominator)
+
+    return NotImplemented
+
+
 class Fraction(numbers.Rational):
     """This class implements rational numbers.
 
     In the two-argument form of the constructor, Fraction(8, 6) will
-    produce a rational number equivalent to 4/3. Both arguments must
-    be Rational. The numerator defaults to 0 and the denominator
-    defaults to 1 so that Fraction(3) == 3 and Fraction() == 0.
+    produce a rational number equivalent to 4/3. The numerator defaults
+    to 0 and the denominator defaults to 1 so that Fraction(3) == 3 and
+    Fraction() == 0.
 
-    Fractions can also be constructed from:
+    Fractions can be constructed from:
```

Comment:
I think you should add the "also" and change the wording here, since this refers to the single-argument constructor:

```suggestion
    Fractions can also be constructed from a single argument, which can be:
```

Reply:
I just reworded this more generally.